src/soundiness.rkt

```diff
@@ -37,14 +38,51 @@
                           num-decrease (length prev)))))
   proof-diffs)
   
+(define (generate-rewrite-once-proof rule loc prog prev)
+  (list (program-body (alt-program prev)) ;; Start Expression
+        (list 'Rewrite=> 
+              (rule-name rule) ;; name
+              (program-body prog)))) ;; expr
 
 (define (add-soundiness-to pcontext ctx altn)
   (match altn
-    [(alt prog `(simplify ,loc ,input #f #f) `(,prev))
-     (define proof (get-proof input
+    ;; This is alt coming from rr
+    [(alt prog `(rr, loc, input #f #f) `(,prev))
+      (cond
+        [(rr-input? input) ;; Check if input is an rr-input struct (B-E-R)
+          (define proof 
+            (get-rr-proof (rr-input-rules input)
+                          (rr-input-input-exprs input) (rr-input-iter-limit input)
+                          (location-get loc (alt-program prev)) (location-get loc prog)))
```

Comment:
Definitely need to clean this up by abstracting over the various proof things.

Reply:
This gets fixed mostly in #560 .